framework/src/st_things/things_stack/framework/things_data_manager.c

```diff
@@ -672,13 +673,13 @@ wifi_manager_softap_config_s *dm_get_softap_wifi_config(void)
 	st_device_s *device = dm_get_info_of_dev(0);
 	int ssid_type = (is_artik == true ? 1 : 0);
 	unsigned char ext_value[9] = { 0, };
-	char ssid_device_name[17];
+	char ssid_device_name[MAX_SSID_DEVICE_NAME + 1] = {0,};
```

Comment:
you want initialize buffer with null string ?
0, Null, \0 they are same value but type is different.
Some compiler doesn't treat 0 as Null string because ascii value is diffrent.
Also snprintf put null string at the end of string, I think you don't have to auto initialize here.

Reply:
oh... Thanks alot.